### DISCUSSION TITLE
Best way to push Quarto website to github repo?

### DISCUSSION THREAD

**Discussion by eteitelbaum** [quarto-dev/quarto-web#294] [Q&A]
What is the best method for pushing your Quarto website to a GitHub repo? Normally I would start a new repo on GitHub, then copy the url and then use version control to start the new project in RStudio. But version control does not have an option for a new Quarto website. So is it better to create the new website using "New Directory" and then push that to GitHub?

**Comment by rgaiacs** [comment]
Hi @eteitelbaum. Thanks for the question. Can you update your installation of RStudio to 2022.07.1 Build 554 as it has the option to use version control for Quarto website?

![Screenshot from 2022-08-02 09-29-35](https://user-images.githubusercontent.com/1506457/182272384-956122b3-c5b7-4f39-9ee6-9347e0cafd1a.png)

**Reply by eteitelbaum** [reply]
Thanks, @rgaiacs. I might not have asked my question in the clearest way. Usually I would create the repo first and then clone it using the version control option for creating a new project in RStudio like this: 

<img width="536" alt="GIT_clone" src="https://user-images.githubusercontent.com/62634082/182386952-61309e25-d544-4156-b670-6eae84946b7a.png">

From there, I would create a new blogdown site in the repo. 

But the only way to create a Quarto site is to create a new project, so that if I did it this way I would have a project within a project. 

So I suppose the best thing is to create the Quarto website while adding a local repo and then push the repo to GitHub. Is that right?

**Comment by rgaiacs** [comment]
Hi @eteitelbaum. To prevent damage any existing project, the "New Project Wizard" will only write the `.Rproj` file in existing project.

> So I suppose the best thing is to create the Quarto website while adding a local repo and then push the repo to GitHub. Is that right?

From RStudio IDE, this is the required workflow at the moment.

**Comment by eteitelbaum** [comment]
Hi @rgaiacs 

> To prevent damage any existing project, the "New Project Wizard" will only write the .Rproj file in existing project.

I didn't get that part. Do you mean to say that the New Project Wizard will only write the .Rproj file to a new project (rather than an existing project)?

**Reply by rgaiacs** [reply]
To a new project in an existing folder. When it creates a new project in a new folder, it will write all the files related with that type of project.
